File: ios/Classes/SelfieData.swift
//
//  SelfieData.swift
//  smart_selfie_ai_model
//
//  Created by Sudarshan Chakra on 23/12/21.
//

import Foundation

struct SelfieData {
    var age: Double
    var gender: String
    var weight: Double
    var height: Double
    
    func toDictionary() -> [String: Any] {
        let bmi = (self.weight / (self.height * self.height)) * 10000
        
        return ["age": self.age.roundToDecimal(2),
                "weight": self.weight.roundToDecimal(2),
                "gender": self.gender,
                "height": self.height.roundToDecimal(2),
                "bmi": bmi.roundToDecimal(2),
               ];
    }
}


extension Double {
    func roundToDecimal(_ fractionDigits: Int) -> Double {
        let multiplier = pow(10, Double(fractionDigits))
        return Darwin.round(self * multiplier) / multiplier
    }
}

File: android/src/main/kotlin/com/sinapsis/smart_selfie_ai_model/SelfieData.kt
package com.sinapsis.smart_selfie_ai_model

import kotlin.math.pow
import kotlin.math.roundToInt

data class SelfieData(val age: Double, val gender: String,
                      val weight: Double, val height: Double)

fun SelfieData.toMap(): HashMap<String, Any?> {
    val bmi = (this.weight / (this.height * this.height)) * 10000
    return hashMapOf(
        "age" to this.age.roundTo(2),
        "weight" to this.weight.roundTo(2),
        "gender" to this.gender,
        "height" to this.height.roundTo(2),
        "bmi" to bmi.roundTo(2)
    )
}

fun Double.roundTo(numFractionDigits: Int): Double {
    val factor = 10.0.pow(numFractionDigits.toDouble())
    return (this * factor).roundToInt() / factor
}